### ISSUE TITLE
Accessibility issues with login form

### ISSUE THREAD

**Issue by selfthinker** [splitbrain/dokuwiki#3457]
A couple of issues on the [login form](https://grimm.haz.wiki/start?do=login&sectok=)...

* [ ] The form fields are related to the user but their [input purpose](https://www.w3.org/TR/WCAG21/#input-purposes) cannot be programmatically determined. It should be set via the `autocomplete` attribute, with `username` for the username and `current-password` for the password. That will help people with cognitive and motor impairments filling them in.
That fails [1.3.5 Identify Input Purpose](https://www.w3.org/TR/WCAG21/#identify-input-purpose).

```
<input type="text" id="focus__this" name="u" class="edit" autocomplete="username">
<input type="password" name="p" class="edit" autocomplete="current-password">
```

* [ ] The error message that appears after giving wrong details ("Sorry, username or password was wrong.") is not programmatically determinable. That means screen reader users won't be alerted to what went wrong.
That fails [4.1.3 Status Messages](https://www.w3.org/TR/WCAG21/#status-messages).

![](https://user-images.githubusercontent.com/108893/113480713-adac9380-948d-11eb-9aec-34ad41cf730a.png)

As this happens on page reload, best move the focus to the error message. As the error message is just a `div`, it should also get a `tabindex=-1` so that it can receive focus.
It makes sense to treat all error messages like that.

## Other non-fail accessibility issues

These don't strictly fail WCAG but are still accessibility issues. Fixing them is recommended but optional for passing WCAG.

* [ ] When you try to log in while using a screen magnifier, get something wrong and the error message appears, it's impossible to notice the message. The same is true for users who zoom into the page or mobile phone users. For those users it will appear as if the form isn't working as the feedback won't be visible to them.
Moving the focus to the error message (the same fix as for the 4.1.3 issue above) would also fix this.

* [ ] The whole login form is difficult to find when using a screen magnifier. That is because everything is centered which creates a big and seemingly empty space on the left of the page.
I suggest to left-align everything. That will be true for other forms as well.
![](https://user-images.githubusercontent.com/108893/113481221-27458100-9490-11eb-9455-1014a946bd00.png)

* [ ] Strictly speaking, you should be able to get to the login page via multiple ways. There is currently only one way to get to it and that is via the link in the header. That link is on every page, so it's less of a problem. But consider adding this (and other application-internal links) to the sitemap/index.
